ci/docker_run.sh

```diff
@@ -17,11 +17,26 @@ fi
 
 echo "Running Docker CI build for '$IMAGE_NAME' "
 
-docker build  -t $IMAGE_NAME .
+# Remove old docker cid just in case
+rm -f docker_cid
+
+docker build -t $IMAGE_NAME .
 exit_code=$?; [[ $exit_code != 0 ]] && exit $exit_code
+
+cleanup() {
+  echo "Running cleanup..."
+  echo "Killing running containers..."
+  cat docker_cid | xargs docker kill 
+  echo "Cleaning stopped docker containers and dangling images/networks/volumes..."
+  docker system prune -f || true
+  echo "Cleanup complete."
+}
+trap cleanup EXIT
```

Comment:
I am not sure how the trap works ... but is it possible move this to the docker_prune script ? IIRC that runs every hour looking for images to clean up.